kornia/geometry/transform/thin_plate_spline.py

```diff
@@ -0,0 +1,148 @@
+
+    # set up and solve linear system
+    # [K   P] [w] = [dst]
+    # [P^T 0] [a]   [ 0 ]
+    pair_distance: torch.Tensor = _pair_square_euclidean(points_src, points_target)
+    k_matrix: torch.Tensor = _kernel_distance(pair_distance)
+
+    zero_mat: torch.Tensor = torch.zeros(batch_size, 3, 3, device=device, dtype=dtype)
+    one_mat: torch.Tensor = torch.ones(batch_size, num_points, 1, device=device, dtype=dtype)
+    dest_with_zeros: torch.Tensor = torch.cat((points_target, zero_mat[:, :, :2]), 1)
+    p_matrix: torch.Tensor = torch.cat((one_mat, points_src), -1)
+    p_matrix_t: torch.Tensor = torch.cat((p_matrix, zero_mat), 1).transpose(1, 2)
+    l_matrix: torch.Tensor = torch.cat((k_matrix, p_matrix), -1)
+    l_matrix = torch.cat((l_matrix, p_matrix_t), 1)
+
+    weights, _ = torch.solve(dest_with_zeros, l_matrix)
+    kernel_weights: torch.Tensor = weights[:, :-3]
+    affine_weights: torch.Tensor = weights[:, -3:]
+
+    return (kernel_weights, affine_weights)
+
+
+def warp_points_tps(points_src: torch.Tensor, kernel_centers: torch.Tensor,
+                    kernel_weights: torch.Tensor, affine_weights: torch.Tensor) -> torch.Tensor:
+    r"""Warp a tensor of coordinate points using the thin plate spline defined by kernel points, kernel weights, and
+    affine weights.
+
+    The source points should be a (B, N, 2) tensor of (x, y) coordinates. The kernel centers are a (B, K, 2) tensor
+    of (x, y) coordinates. The kernel weights are a (B, K, 2) tensor, and the affine weights are a (B, 3, 2) tensor.
+    For the weight tensors, tensor[..., 0] contains the weights for the x-transform and tensor[..., 1] the weights
+    for the y-transform.
+
+    Returns a (B, N, 2) tensor containing the warped source points.
+
+    Args:
+        points_src (torch.Tensor): tensor of source points (B, N, 2)
+        kernel_centers (torch.Tensor): tensor of kernel center points (B, K, 2)
+        kernel_weights (torch.Tensor): tensor of kernl weights (B, K, 2)
+        affine_weights (torch.Tensor): tensor of affine weights (B, 3, 2)
+
+    Returns:
+        torch.Tensor: The warped source points, from applying the TPS transform.
+    """
+    # f_{x|y}(v) = a_0 + [a_x a_y].v + \sum_i w_i * U(||v-u_i||)
+    pair_distance: torch.Tensor = _pair_square_euclidean(points_src, kernel_centers)
+    k_matrix: torch.Tensor = _kernel_distance(pair_distance)
+
+    # broadcast the kernel distance matrix against the x and y weights to compute the x and y
+    # transforms simultaneously
+    warped: torch.Tensor = (
+        k_matrix[..., None].mul(kernel_weights[:, None]).sum(-2) +
+        points_src[..., None].mul(affine_weights[:, None, 1:]).sum(-2) +
+        affine_weights[:, None, 0]
+    )
+
+    return warped
+
+
+def warp_img_tensor_tps(tensor: torch.Tensor, kernel_centers: torch.Tensor, kernel_weights: torch.Tensor,
+                        affine_weights: torch.Tensor, align_corners: bool = False) -> torch.Tensor:
```

Comment:
Function name too long I think. We could find something short and clean, what about `warp_image_tps` - it's close to opencv [`warpImage`](https://docs.opencv.org/3.4/df/dfe/classcv_1_1ShapeTransformer.html). We should check other libs how they name this.

Reply:
/cc @ducha-aiki @shijianjian @lferraz @anguelos @pmeier

Reply:
I'm changing it to `warp_image_tps`. Any alternatives someone prefers?